src/Umbraco.Core/PropertyEditors/BlockGridConfiguration.cs

```diff
@@ -0,0 +1,140 @@
+                
+                [DataMember(Name ="columnSpan")]
+                public int? ColumnSpan { get; set; }
+
+                [DataMember(Name ="rowSpan")]
+                public int? RowSpan { get; set; }
+
+                [DataMember(Name ="minAllowed")]
+                public int? MinAllowed { get; set; }
+                
+                [DataMember(Name ="maxAllowed")]
+                public int? MaxAllowed { get; set; }
+
+                [DataMember(Name ="onlySpecifiedAllowance")]
+                public bool? OnlySpecifiedAllowance { get; set; }
+
+                [DataMember(Name ="specifiedAllowance")]
+                public BlockGridAreaConfigurationSpecifiedAllowance[] SpecifiedAllowance { get; set; }  = null!;
+
+                [DataContract]
+                public class BlockGridAreaConfigurationSpecifiedAllowance
+                {
+                    [DataMember(Name ="elementTypeKey")]
+                    public Guid? ElementTypeKey { get; set; }
+
+                    /*
+                    [DataMember(Name ="groupKey")]
+                    public Guid? groupKey { get; set; }
+                    */
+
+                    [DataMember(Name ="minAllowed")]
+                    public int? MinAllowed { get; set; }
+                    
+                    [DataMember(Name ="maxAllowed")]
+                    public int? MaxAllowed { get; set; }
+
+                }
+
+            }
+
+            [DataMember(Name ="backgroundColor")]
+            public string? BackgroundColor { get; set; }
+
+            [DataMember(Name ="iconColor")]
+            public string? IconColor { get; set; }
+
+            [DataMember(Name ="thumbnail")]
+            public string? Thumbnail { get; set; }
+
+            [DataMember(Name ="contentElementTypeKey")]
+            public Guid ContentElementTypeKey { get; set; }
+
+            [DataMember(Name ="settingsElementTypeKey")]
+            public Guid? SettingsElementTypeKey { get; set; }
+
+            [DataMember(Name ="view")]
+            public string? View { get; set; }
+
+            [DataMember(Name ="stylesheet")]
+            public string? Stylesheet { get; set; }
+
+            [DataMember(Name ="label")]
+            public string? Label { get; set; }
+
+            [DataMember(Name ="editorSize")]
+            public string? EditorSize { get; set; }
+
+            [DataMember(Name ="forceHideContentEditorInOverlay")]
+            public bool ForceHideContentEditorInOverlay { get; set; }
+        }
+
+        [ConfigurationField("validationLimit", "Amount", "numberrange", Description = "Set a required range of root blocks")]
+        public NumberRange ValidationLimit { get; set; } = new NumberRange();
+
+        [DataContract]
+        public class NumberRange
+        {
+            [DataMember(Name ="min")]
+            public int? Min { get; set; }
+
+            [DataMember(Name ="max")]
+            public int? Max { get; set; }
+        }
```

Comment:
@bergmania could we re-use the `NumberRange` class bound to the prevalue editor?
https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Core/PropertyEditors/BlockListConfiguration.cs#L59-L67

It would also make it easier for package developer to re-use this class and the view for the prevalue editor.

Reply:
@nielslyngsoe @kjac can we move the `NumberRange` to a central location, which can be shared between Block List and Block Grid editor, but also used by developers creating custom block based editors?

Reply:
@bjarnef unfortunately not; they are public classes, nested within other public classes (`MediaPicker3Configuration`, `BlockListConfiguration`). If we were to make it shared, we would have to move it to a shared location (i.e. outside of `BlockListConfiguration`), which would be a breaking change.

We could schedule this change for V12+, but we would still have to use the nested `NumberRange` implementations until then, which would leave an unused shared `NumberRange` implementation hanging until then. It would be somewhat strange, and I'm not sure it's entirely worth it? I know that means encouraging code duplication for other usages of a range implementation, but the solution just feels ... clunky?

On a side note, the `NumberRange` implementations are pretty limited, whereas the `Range<T>` (in `Umbraco.Cms.Core.Models`) might be somewhat more useful to other people's implementation needs. There is also the `Range` struct available in `System`.

Reply:
@kjac okay I see, that sucks. I guess it isn't the last editor that need the min/max limit in the configuration, e.g. it MTNP and Multiple URL Picker later would use same configuration for consistency (but would probably be in a new editor or some migrations needed to handle existing separate min/max properties) - in that case we would have 4, 5 or more `NumberRange` classes.

It may be worth considering this in future versions though. There could probably also be a class in core to serialize the default BlockList model (and now BlockGrid), when creating blocklist/blockgrid content programmatically.
https://our.umbraco.com/Documentation/Fundamentals/Backoffice/property-editors/built-in-property-editors/Block-List-Editor/#creating-blocklist-programmatically